dronegen/container_image_triggers.go

```diff
@@ -0,0 +1,159 @@
+// Copyright 2021 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"path"
+)
+
+// Describes a Drone trigger as it pertains to container image building.
+type TriggerInfo struct {
+	Trigger           trigger
+	Name              string
+	Flags             *TriggerFlags
+	SupportedVersions []*ReleaseVersion
+	SetupSteps        []step
+}
+
+// This is mainly used to make passing these vars around cleaner
+type TriggerFlags struct {
+	ShouldAffectProductionImages bool
+	ShouldBuildNewImages         bool
+	UseUniqueStagingTag          bool
+}
+
+func NewTagTrigger(branchMajorVersion string) *TriggerInfo {
```

Comment:
Nit: Can you add godocs on all public methods with some explanations on what they're for and what they do? E.g. `NewTagTrigger creates a new trigger that will execute when a new tag is pushed to Teleport repository, build multi-arch container images and push them to staging repositories`, something like that (feel free to tweak for accuracy).